pennylane/workflow/return_types_spec.rst

```diff
@@ -0,0 +1,196 @@
+>>> result = qml.device('default.qubit').execute(tape)
+>>> result
+[{'0': 50}, {'0': 46, '1': 4}, {'0': 32, '1': 18}]
+>>> batch, fn = qml.transforms.broadcast_expand(tape)
+>>> fn(qml.device('default.qubit').execute(batch))
+array([{'0': 50}, {'0': 39, '1': 11}, {'0': 28, '1': 22}], dtype=object)
+
+
+Single Tape
+-----------
+
+If the tape has a single measurement, then the result corresponding to that tape simply obeys the specification
+above.  Otherwise, the result for a single tape is a ``tuple`` where each entry corresponds to each
+of the corresponding measurements. In the below example, the first entry corresponds to the first
+measurement process ``qml.expval(qml.Z(0))``, the second entry corresponds to the second measurement process
+``qml.probs(wires=0)``, and the third result corresponds to the third measurement process ``qml.state()``.
+
+>>> tape = qml.tape.QuantumScript((), (qml.expval(qml.Z(0)), qml.probs(wires=0), qml.state()))
+>>> qml.device('default.qubit').execute(tape)
+(1.0, array([1., 0.]), array([1.+0.j, 0.+0.j]))
+
+**Shot vectors:**
+
+When a shot vector is present ``shots.has_partitioned_shot``, the measurement instead becomes a
+tuple where each entry corresponds to a different shot value.
+
+>>> measurements = (qml.expval(qml.Z(0)), qml.probs(wires=0))
+>>> tape = qml.tape.QuantumScript((), measurements, shots=(50,50,50))
+>>> result = qml.device('default.qubit').execute(tape)
+>>> result
+((1.0, array([1., 0.])), (1.0, array([1., 0.])), (1.0, array([1., 0.])))
+>>> result[0]
+(1.0, array([1., 0.]))
+>>> tape = qml.tape.QuantumScript((), [qml.counts(wires=0)], shots=(1, 10, 100))
+>>> qml.device('default.qubit').execute(tape)
+({'0': 1}, {'0': 10}, {'0': 100})
+
+Let's look at an example with all forms of nesting.  Here, we have a tape with a batch size of ``3``, three
+diferent measurements with different fundamental shapes, and a shot vector with three different values.
+
+>>> op = qml.RX((1.2, 2.3, 3.4), 0)
+>>> ms = (qml.expval(qml.Z(0)), qml.probs(wires=0), qml.counts())
+>>> tape = qml.tape.QuantumScript((op,), ms, shots=(1, 100, 1000))
+>>> result = qml.device('default.qubit').execute(tape)
+>>> result
+((array([ 1., -1., -1.]),
+array([[1., 0.],
+       [0., 1.],
+       [0., 1.]]),
+[{'0': 1}, {'1': 1}, {'1': 1}]),
+(array([ 0.3 , -0.66, -0.98]),
+array([[0.61, 0.39],
+       [0.13, 0.87],
+       [0.03, 0.97]]),
+[{'0': 61, '1': 39}, {'0': 13, '1': 87}, {'0': 3, '1': 97}]),
+(array([ 0.364, -0.648, -0.962]),
+array([[0.669, 0.331],
+       [0.165, 0.835],
+       [0.012, 0.988]]),
+[{'0': 669, '1': 331}, {'0': 165, '1': 835}, {'0': 12, '1': 988}]))
+
+
+>>> result[0][0] # first shot value, first measurement
+array([ 1., -1., -1.])
+>>> result[0][0][0] # first shot value, first measurement, and parameter of 1.2
+1.0
+>>> result[1][2] # second shot value, third measurement, all three parameter values
+[{'0': 74, '1': 26}, {'0': 23, '1': 77}, {'1': 100}]
+
+
+Mid-circuit measurements
```

Comment:
Minor detail, but what should the header level for this be? Right now it renders as a sub-heading to "Single Tape", but a "bigger" subheading than "Shot vectors", so it gets its own entry in the TOC, and none of the others do:

![image](https://github.com/PennyLaneAI/pennylane/assets/38584660/140c480a-3fde-4ad1-a4ac-794a1ec0efbf)

Reply:
Potentially I can make things like shot vectors and broadcasting heading as well, instead of just bolded words.